modules/identities/scripts/link.ts

```diff
@@ -0,0 +1,45 @@
+import { RuntimeError, ScriptContext } from "../module.gen.ts";
+import { IdentityDataInput, IdentityProviderInfo } from "../utils/types.ts";
+
+export interface Request {
+	userToken: string;
+    info: IdentityProviderInfo;
+    uniqueData: IdentityDataInput;
+    additionalData: IdentityDataInput;
+}
+
+export interface Response {
+	identityProviders: IdentityProviderInfo[];
+}
+
+export async function run(
+	ctx: ScriptContext,
+	req: Request,
+): Promise<Response> {
+	await ctx.modules.rateLimit.throttle({
+        key: req.userToken,
+        period: 10,
+        requests: 10,
+        type: "user",
+    });
+
+    // Ensure the user token is valid and get the user ID
+    const { userId } = await ctx.modules.users.authenticateToken({ userToken: req.userToken } );
```

Comment:
see comment in list

Reply:
I'll do that for this one— I'll wait for your comments on whether it makes sense on `list`.